### PR TITLE
Ident on paste - bunch of fixes

### PR DESCRIPTION
**Pull request by dos65**
There were several issues with ident on paste that is fixed by this
change:
- weird formatting if second line is empty of  overindented:
  ```scala
  // original
  trait X[A] {
      def foo: X[A] // 4 spaces instead of 2
  }
  // pasted
  object Main {
    trait X[A] {
      def foo: X[A]
  }                // <- wrong formatted
  }
  ```
- previous logic were removing all empty lines that wasn't convinient if
  you copy a large block of code like:
  ```scala
  // original
  trait X[A] {
    def foo: X[A]
  }

  def m1: Int = 32

  // pasted without whitespace between `X` and `m1`
  trait X[A] {
    def foo: X[A]
  }
  def m1: Int = 32
  ```
- normalization from tabs to spaces didn't work correctly

### CODE REVIEW

**Review by tgodzik — CHANGES_REQUESTED** [review]
Looks good! Just two minor comments from my side.

File: tests/unit/src/test/scala/tests/rangeFormatting/IndentWhenPastingSuite.scala [hunk 1]
```diff
@@ -174,9 +180,9 @@ class IndentWhenPastingSuite
        |object Main:
        |  object SubMain:
        |    enum                        Color(val rgb: Int):
-       |      case Red   extends Color(0xFF0000)
-       |      case Green extends Color(0x00FF00)
-       |      case Blue  extends Color(0x0000FF)
+       |       case Red   extends Color(0xFF0000)
```
[the remark this answers is not in the document]
Comment (dos65) [reply tests/unit/src/test/scala/tests/rangeFormatting/IndentWhenPastingSuite.scala:183]: In pasted code it was 3.
Previously it was corrected because of implementations specifics but it was rather a corner case than general logic.

File: metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala [hunk 2]
```diff
@@ -24,35 +24,30 @@ object IndentOnPaste extends RangeFormatter {
...
-    } else line
+    import opts._
+    codeStartPosition(line).filter(_ > 0) match {
+      case Some(codeStartPos) =>
+        val (indentation, code) = line.splitAt(codeStartPos)
+        val pastedBlank = indentation.head
+        blank match {
+          case _ if pastedBlank == blank => line
+          case '\t' if pastedBlank == ' ' =>
+            val tabNum = math.ceil(indentation.length.toDouble / 2).toInt
```
[the remark this answers is not in the document]
Comment (dos65) [reply metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala:43]: Hmm, I realized that current implementation doesn't work well with tabs not only here. Will do an additional changes for tabs

[the remark this answers is not in the document]
Comment (dos65) [reply metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala:43]: The previous comment was wrong.

Spaces to tabs convertation is a tricky one.
If user has `tabSize` larger than 2 then pasting code that was taken outside of his project might be realigned incorrectly. 

Let's keep it 2 here for now and see if there will be any issue about that. 
Honestly, I don't think that there will be any as scalafmt doesn't support tabs :smile:

[the remark this answers is not in the document]
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala:43]: ok sure

File: tests/unit/src/test/scala/tests/rangeFormatting/IndentWhenPastingSuite.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/unit/src/test/scala/tests/rangeFormatting/IndentWhenPastingSuite.scala:183]: Pasted code was 3, but we adjsut it to fit how the current document is formatted.

[the remark this answers is not in the document]
Comment (dos65) [reply tests/unit/src/test/scala/tests/rangeFormatting/IndentWhenPastingSuite.scala:183]: We shouldn't touch it because it's relative indentation in pasted block .
The only we could do in this formatter is to increase/decrease indentation for the whole pasted block.

Examples:
```scala
// pasted code
Color:
     case Red extends Color(0xFF0000)

// paste - increase indentation by 2 on each line
object Main: 
  Color:                                                   
      case Red extends Color(0xFF0000)
     
// or if pasted block was
      Color:
           case Red extends Color(0xFF0000)

// paste - decrease indetation by 4 on each line
object Main: 
  Color:                                                    
      case Red extends Color(0xFF0000)
      
// we should keep the original indentation between pasted lines
// otherwise we might drop significant indentation or uglify code
val a =
  2

// paste -  how it shouldn't work
object Main:
  val a = 
  2          
```

[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/unit/src/test/scala/tests/rangeFormatting/IndentWhenPastingSuite.scala:183]: I am not talking here about dropping the indentation, but rather if we detect the diff in indentation between consecutive lines in the pasted fragment to be different than the current indentation, we could change it to the current indentation.

[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/unit/src/test/scala/tests/rangeFormatting/IndentWhenPastingSuite.scala:183]: I think that was the intention before, that we would adjust the indentation.

[the remark this answers is not in the document]
Comment (dos65) [reply tests/unit/src/test/scala/tests/rangeFormatting/IndentWhenPastingSuite.scala:183]: Previously it worked only for the second line in this case because of implementation specifics. 

For example if you paste this sample into current implementation - the second line indent will be reduced  from 3 to 2 but the third one keep the relative difference 3. 
<pre><code>
Color:
&nbsp;&nbsp;&nbsp;case Red
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;case Black
</code></pre>

I would prefer not to try to correct relative indentation here.
It seems that it was a reason why multiline string pasting wasn't perfect in this case (now it's fixed): https://github.com/scalameta/metals/blob/2adcb3c47e923cd78492e3845d701ef3b9600fb1/tests/unit/src/test/scala/tests/rangeFormatting/MultilineStringRangeFormattingWhenPastingSuite.scala#L330-L332

**Review by tgodzik — APPROVED** [review]
LGTM! Thanks for fixing this and being patient with my questions :tada:
